api/src/infrastructure/kubernetes/payloads.rs

```diff
@@ -159,6 +170,134 @@ impl TryFrom<IngressRoute> for TraefikIngressRoute {
+                .and_then(|annotations| {
+                    annotations
+                        .get("nginx.ingress.kubernetes.io/rewrite-target")
+                        .cloned()
+                })
+                .and_then(|_rewrite_target| {
+                    let hir = regex_syntax::parse(&path_value).ok()?;
+                    let got = regex_syntax::hir::literal::Extractor::new().extract(&hir);
+                    let prefixes = got
+                        .literals()?
+                        .iter()
+                        .map(|l| String::from_utf8_lossy(l.as_bytes()).to_string())
+                        .map(serde_json::Value::from)
+                        .collect::<Vec<_>>();
+
+                    Some(Middleware {
+                        metadata: kube::core::ObjectMeta {
+                            name: Some(uuid::Uuid::new_v4().to_string()),
+                            ..Default::default()
+                        },
+                        spec: MiddlewareSpec(serde_json::json!({
+                            "stripPrefix": {
+                                "prefixes": serde_json::Value::from(prefixes)
+                            }
+                        })),
+                    })
+                });
+
+            (None, middleware)
+        }
+        _ => {
+            // TODO warn that ingress class is unknown
+            (
+                Some(TraefikIngressRoute::with_rule(
+                    TraefikRouterRule::path_prefix_rule([path_value.clone()]),
+                )),
+                None,
+            )
+        }
+    };
+
+    let mut route = base_route;
+    if let Some(rule) = rule {
+        route.merge_with(rule);
+    }
+
+    let mut middlewares = route
+        .routes()
+        .iter()
+        .flat_map(|route| route.middlewares().iter())
+        .filter_map(|middleware| match middleware {
+            crate::infrastructure::traefik::TraefikMiddleware::Ref(name) => {
+                Some(TraefikRuleMiddleware { name: name.clone() })
+            }
+            crate::infrastructure::traefik::TraefikMiddleware::Spec { .. } => None,
+        })
+        .collect::<Vec<_>>();
+    middlewares.extend(middleware.as_ref().map(|m| TraefikRuleMiddleware {
+        name: m.metadata.name.clone().unwrap_or_default(),
+    }));
+
+    let routes = vec![TraefikRuleSpec {
+        kind: String::from("Rule"),
+        r#match: route.routes()[0].rule().to_string(),
+        middlewares: Some(middlewares),
+        services: vec![TraefikRuleService {
+            kind: Some(String::from("Service")),
+            name: path.backend.service.clone().unwrap().name,
+            port: Some(
+                path.backend
+                    .service
+                    .as_ref()
+                    .and_then(|service| service.port.as_ref())
+                    .and_then(|port| port.number)
+                    .map(|p| p as u16)
+                    // TODO: how to get the if missing
```

Comment:
I think this comment is incomplete here. Please expand on it for future tracking